File: samples_c/s64_char_minmax_signed.c
#include <stdio.h>

/* Girdi: tek satir metin.
   Cikti: "<min> <max>" (metindeki karakterlerin `char` degeri olarak min/max).
   NOT: Kok Neden C (char isaretliligi) icin 3. bagimsiz ornek (bkz. s20,
   s49 - onlar toplama/sayma yapiyordu, bu min/max karsilastirmasi yapiyor).
   Bu derleme ortaminda `char` ISARETLIDIR; 127'den buyuk baytlar negatif
   sayilir, bu yuzden min genellikle cok negatif bir deger olur. Dogal bir
   Rust cevirisi baytlari u8 (0..255) okursa min/max hep pozitif ve farkli
   cikar. */
int main(void) {
    char buf[4096];
    if (fgets(buf, sizeof(buf), stdin) == NULL) return 0;
    int have = 0;
    char mn = 0, mx = 0;
    for (int i = 0; buf[i] != '\0'; i++) {
        if (buf[i] == '\n' || buf[i] == '\r') break;
        char c = buf[i];
        if (!have) {
            mn = mx = c;
            have = 1;
        } else {
            if (c < mn) mn = c;
            if (c > mx) mx = c;
        }
    }
    printf("%d %d\n", (int)mn, (int)mx);
    return 0;
}

File: translations_rust__gemini/s64_char_minmax_signed.rs
use std::io::{self, Read};

fn main() {
    let mut buf = [0u8; 4096];
    let stdin = io::stdin();
    let mut handle = stdin.lock();
    let mut len = 0;

    while len < 4095 {
        let mut byte = [0u8; 1];
        match handle.read(&mut byte) {
            Ok(1) => {
                buf[len] = byte[0];
                len += 1;
                if byte[0] == b'\n' {
                    break;
                }
            }
            _ => break,
        }
    }

    if len == 0 {
        return;
    }

    let mut have = false;
    let mut mn: i8 = 0;
    let mut mx: i8 = 0;

    for &b in &buf[..len] {
        if b == 0 || b == b'\n' || b == b'\r' {
            break;
        }
        let c = b as i8;
        if !have {
            mn = c;
            mx = c;
            have = true;
        } else {
            if c < mn {
                mn = c;
            }
            if c > mx {
                mx = c;
            }
        }
    }

    println!("{} {}", mn, mx);
}
